### PR TITLE
Sync Paydock with PowerBoard

### PR DESCRIPTION
**Pull request by FabianaRBarros**

### CODE REVIEW

**Review by andryes — DISMISSED** [review]

File: includes/Abstracts/AbstractWalletBlock.php [hunk 1]
```diff
@@ -34,6 +33,7 @@ public function get_payment_method_data(): array {
 			'total_limitation' => $settings->getWidgetPaymentWalletMinMax( $payment ),
```
Comment (Danyl-Kushnaryov) [inline includes/Abstracts/AbstractWalletBlock.php:33]: @andryes  @FabianaRBarros Just FYI I don't know if it's correct but in the `power_board` branch this method is not present: `getWidgetPaymentWalletMinMax`

File: includes/Controllers/Admin/WidgetController.php
Comment (Danyl-Kushnaryov) [inline includes/Controllers/Admin/WidgetController.php:1]: @FabianaRBarros @andryes  I'm seeing also such a difference in this file (`power_board` is on the left, `main` is on the right)

Screenshot 2024-08-29 at 16 30 20

File: includes/Abstracts/AbstractWalletBlock.php (hunk 1 above)
Comment (FabianaRBarros) [reply includes/Abstracts/AbstractWalletBlock.php:33]: This is something that was present on `main` branch and not on `powerboard`. I check and it was added by jsp [here](https://github.com/PayDock/e-commerce-woo/commit/37d99e953b4646579f5fdf793a36931049cc62c1), so I assumed it should not be removed as it was a recent implementation.

Comment (FabianaRBarros) [reply includes/Abstracts/AbstractWalletBlock.php:33]: Maybe @andryes and @jpcostello have more information about this. Should we keep or remove?

File: resources/js/frontend/index.js [hunk 2]
```diff
@@ -231,7 +268,6 @@ const Content = (props) => {
         checkboxSavedCardsComponent(labels.saveCardLabel)
     );
 };
-
 const Label = (props) => {
```
Comment (Danyl-Kushnaryov) [inline resources/js/frontend/index.js:271]: @FabianaRBarros  It looks like it's not present in the `power_board` and in the branch `sync-main-with-power-board` this variable is not being used. So maybe it should be removed from your branch?

Comment (Danyl-Kushnaryov) [reply resources/js/frontend/index.js:271]: Screenshot 2024-09-04 at 15 30 54

File: resources/js/includes/apms.js
Comment (Danyl-Kushnaryov) [inline resources/js/includes/apms.js:1]: The same for apms. It looks like this Label is not being used and can be removed.

Screenshot 2024-09-04 at 15 37 29

Screenshot 2024-09-04 at 15 38 21

File: includes/Abstracts/AbstractWalletBlock.php (hunk 1 above)
Comment (Danyl-Kushnaryov) [reply includes/Abstracts/AbstractWalletBlock.php:33]: Yes, or maybe we should add this to the `power_board` :)

Comment (jpcostello) [reply includes/Abstracts/AbstractWalletBlock.php:33]: @FabianaRBarros @Danyl-Kushnaryov This line can be removed - min/max checks should only take place for APMs (AbstractAPMsBlock)

File: includes/Controllers/Webhooks/PaymentController.php [hunk 3]
```diff
@@ -19,12 +24,15 @@ public function capturePayment() {
...
 			$order = wc_get_order( $orderId );
-			if ( 'paydock-authorize' != $order->get_status() ) {
-				$error = __( 'The order should be have status "paydock-authorize"', 'woocommerce' );
-			}
+			/*if ( ! in_array( $order->get_meta( ActivationHook::CUSTOM_STATUS_META_KEY ), [
+				'paydock-authorize',
+				'wc-paydock-authorize'
+			] ) ) {
+				$error = __( 'The order has been authorized and is awaiting approval.', 'paydock'  );
+			}*/
```
Comment (jpcostello) [inline includes/Controllers/Webhooks/PaymentController.php:35]: Can we remove comments?

File: includes/Controllers/Webhooks/PaymentController.php [hunk 4]
```diff
@@ -34,36 +42,38 @@ public function capturePayment() {
...
 
-		if ( ! empty( $orderTotal ) ) {
-			$amount = $orderTotal;
-		} else {
+		// if ( ! empty( $orderTotal ) ) {
+		// 	$amount = $orderTotal;
+		// } else {
 			$amount = wc_format_decimal( $_POST['amount'] );
-		}
+		// }
```
Comment (jpcostello) [inline includes/Controllers/Webhooks/PaymentController.php:49]: Can we remove comments?

File: assets/js/frontend/form.js [hunk 5]
```diff
@@ -177,10 +271,26 @@ setTimeout(() => jQuery(function ($) {
...
         }
 
+        /*
         if (paydockCardSettings.hasOwnProperty('styles') && paydockCardSettings.cardSupportedCardTypes !== '') {
             supportedCard = paydockCardSettings.cardSupportedCardTypes.replaceAll(' ', '').split(',')
-            widget.setSupportedCardIcons(supportedCard);
+
+            widget.setSupportedCardIcons(supportedCard, true);
         }
+        */
```
Comment (jpcostello) [inline assets/js/frontend/form.js:280]: Can we remove comments?

File: includes/Services/ActionsService.php [hunk 6]
```diff
@@ -40,6 +39,24 @@ protected function __construct() {
 			$this->addEndpoints();
 			$this->addOrderActions();
 		} );
+
+		add_filter( 'gettext', array( $this, 'cba_refund_msg' ), 20, 3 );
+	}
+
+	public function cba_refund_msg( $translated_text, $text, $domain ) {
```
Comment (jpcostello) [inline includes/Services/ActionsService.php:46]: Make function name generic

File: includes/Services/OrderService.php [hunk 7]
```diff
@@ -105,23 +127,23 @@ public function statusChangeVerification( $orderId, $oldStatusKey, $newStatusKey
...
+		if ( is_object( $order ) ) {
+
+			$capturedAmount = $order->get_meta( 'capture_amount' );
+
+			if ( ! empty( $capturedAmount ) ) {
+
+				// if ( $order->get_total() > $capturedAmount ) {
+					$this->templateService->includeAdminHtml( 'information-about-partial-captured', compact( 'order', 'capturedAmount' ) );
+				// }
+
```
Comment (jpcostello) [inline includes/Services/OrderService.php:142]: Remove comments?

**Review by jpcostello — CHANGES_REQUESTED** [review]

**Review by andryes — APPROVED** [review]

**Review by jpcostello — APPROVED** [review]

**Pull request merged**